src/lib/wallet.ts

```diff
@@ -480,17 +483,32 @@ export const usePublishedHistory = (path: string) => {
       const iterable: AsyncIterable<Record<string, unknown>> =
         await follower.getReverseIterable();
       setStatus(LoadStatus.Waiting);
+
       const items = [];
+      let fetchNextP = new Promise<void>(res =>
+        setFetchNextPage(() => () => res()),
+      );
+      let curPageSize = 0;
+
       for await (const { value } of iterable) {
+        if (paginationSize && curPageSize >= paginationSize) {
+          setData(items);
+          await fetchNextP;
+          fetchNextP = new Promise<void>(res =>
```

Comment:
this is a little hard for me to follow. `res` resolves when the effect triggered by `setFetchNextPage` completes?

Could it work with a counter of pages fetched? That might be a clearer state to trigger

Reply:
I tried doing something more iterative with `iterable[Symbol.asyncIterator]()` but it was more messy and I gave up working through bugs. Just added some comments even though this is a little tricky with the promises.